### PR TITLE
Rewrite Sparse Block bulkAdd

### PR DESCRIPTION
**Pull request by e-kayrakli**
This is a small PR that updates bulkAdd with the help of new comparator support in Search module.

Particularly, bulkAdd helper in SparseBlockDom was using a linear search to sift through indices and separate them according to their target locales. With this update, a dummy pair of start-end indices created for each locale and they are searched in the inds array using binary search. Roughly speaking, we now spend 2*numLocales*O(logn) instead of O(n) time for finding targetLocales.

Passed standard and GASNet configs.

### CODE REVIEW

File: test/users/engin/sparse_bulk_dist/sparseAdd2D.chpl [hunk 1]
```diff
@@ -7,7 +7,7 @@ const FullDom = {0..#N, 0..#N} dmapped Block({0..#N, 0..#N});
 
 config type layoutType = DefaultDist;
 var layout = new layoutType;
-var FullSparseDom: sparse subdomain(FullDom) dmapped new dmap(layout);
+var FullSparseDom: sparse subdomain(FullDom);
```
Comment (mppf) [inline test/users/engin/sparse_bulk_dist/sparseAdd2D.chpl]: These test changes look unrelated to what you described in the PR message. I don't mind if they go in the same PR but the PR message should at least say it also removes some no-longer-necessary code in tests, assuming that's what's going on.

**Comment by mppf** [comment]
I like it. You'll have to resolve merge conflicts, do testing, and possibly commit the test changes in a different PR, but you can consider it reviewed. Thanks!

**Comment by mppf** [comment]
Ah, you've already done testing. Probably just fix the merge conflicts and update the PR message to describe what's going on with the tests then.

**Comment by e-kayrakli** [comment]
I forgot that I had made those changes in another branch that is now in master.(Which was the reason of merge conflicts) With the last revision there is no test changes specific to this PR.

Rerunning some relevant tests to be sure, and then will merge.
